### DISCUSSION TITLE
Zigbee2MQTT Zigbee network performance troubleshooting

### DISCUSSION THREAD

**Discussion by dreamstack-net** [Koenkk/zigbee2mqtt#22180] [General]
I'm trying to troubleshoot my HA / Zigbee2MQTT setup. The issue I'm experiencing is that most light switches only work 50/50. Another symptom is that only a few lights will turn on or off instead of all (GU10 lights in a single lamp. I did create groups in HA, but it still often). I'm early in the troubleshooting process, and I'm relatively new to Zigbee2MQTT, but this is what I have done so far:

- I looked closely at the troubleshooting doc on the Z2M site, and all the things mentioned there are either already in place or irrelevant to my setup.
- I use a QinHeng Electronics SONOFF Zigbee 3.0 USB Dongle Plus V2 dongle (model E). I've upgraded the firmware to 7.3.1.0 build 176 (the latest version, I think)
- I looked in the logs, and there are no errors.
- I generated a map and looked at it. Network strength/coverage shouldn't be a problem; all devices should have a strong connection.

When I looked at the map, I was surprised by the (in my eyes) absurd number of connections between all routers. I understand Zigbee networks are mesh networks, and that's the power, but could network congestion be an issue here, see image? (https://imgur.com/a/ZCSVp9m)

Most of my lights that have issues are Hue lights (I don't think it's Hue-related; most of my lights are Hue lights), and all my smart switches are Hue wall switches.

I configured some HUE wall switches with a single rocker in Z2M as a direct binding, and they always work fine, but dual rocker switches are not supported, so that needs to go via HA.

Switching lights on via my phone usually works fine, even when the switches are acting up.

I'm sure I'm missing something obvious, but what would be the next step in my troubleshooting journey?

**Comment by diegocjorge** [comment]
> I'm trying to troubleshoot my HA / Zigbee2MQTT setup. The issue I'm experiencing is that most light switches only work 50/50. Another symptom is that only a few lights will turn on or off instead of all (GU10 lights in a single lamp. I did create groups in HA, but it still often). I'm early in the troubleshooting process, and I'm relatively new to Zigbee2MQTT, but this is what I have done so far:

You should create groups in Zigbee2MQTT instead of creating them in Home Assistant.


> When I looked at the map, I was surprised by the (in my eyes) absurd number of connections between all routers. I understand Zigbee networks are mesh networks, and that's the power, but could network congestion be an issue here, see image? (https://imgur.com/a/ZCSVp9m)

It's fine, nothing out of the ordinary.

> I configured some HUE wall switches with a single rocker in Z2M as a direct binding, and they always work fine, but dual rocker switches are not supported, so that needs to go via HA.

From my understanding you have 2 Zigbee networks? Some devices are connected to Zigbee2MQTT and the rest you have them in the HUE bridge that is integrated in HA?

Check if they are not broadcasting in the same channel and causing interference with each other.

**Reply by dreamstack-net** [reply]
Thanks for your reply,

>You should create groups in Zigbee2MQTT instead of creating them in Home Assistant.

Ah, that's a good tip. I've played with Zigbee2MQTT groups but eventually created them in HA; I'll create them in Zigbee2MQTT and call these groups in my HA automations instead.

>It's fine, nothing out of the ordinary.

That would rule out my network congestion theory. 

>From my understanding, you have 2 Zigbee networks. Some devices are connected to Zigbee2MQTT, and the rest are in the HUE bridge that is integrated in HA.

No, I only have a Zigbee2MQTT Zigbee network, all devices are in Zigbee2MQTT, I don't have any other (like HUE) Zigbee networks. 

I'll play around with it this weekend to see if I can make any improvements with the Zigbee2MQTT groups.
